CMake error on Summit with default IBM compiler when compile_threaded is true

This issue is reproducible on Summit with latest E3SM master branch.

Steps to reproduce:
```
git clone https://github.com/E3SM-Project/E3SM.git
cd E3SM
git submodule update --init
cd cime/scripts
./create_newcase --case X_f19_g16 --compset X --res f19_g16
cd X_f19_g16
./xmlchange NTHRDS_ATM=2
./case.setup
./case.build
```
The error information is:
```
...
CMake Error at cmake/cmake_util.cmake:53 (message):
  Trying to set flags on non-existent source:
  share/util/shr_reprosum_mod.F90, looked for
  components/share/util/shr_reprosum_mod.F90
Call Stack (most recent call first):
  cime/scripts/X_f19_g16/Depends.ibm.cmake:11 (e3sm_add_flags)
  cmake/build_model.cmake:150 (include)
  cmake/atm/CMakeLists.txt:1 (build_model)
...
```
The non-existent source is from cime/config/e3sm/machines/Depends.ibm.cmake:
```
if (compile_threaded)
  e3sm_add_flags("share/util/shr_reprosum_mod.F90" "-qsmp=noauto:noomp")
endif()
```
The path share/util/shr_reprosum_mod.F90 might need to be modified to something like below:
${CIMESRC_PATH}/share/util/shr_reprosum_mod.F90

--- a/cime/config/e3sm/machines/Depends.ibm.cmake
+++ b/cime/config/e3sm/machines/Depends.ibm.cmake
@@ -8,7 +8,7 @@ foreach(ITEM IN LISTS SSOBJS)
 endforeach()
 
 if (compile_threaded)
-  e3sm_add_flags("share/util/shr_reprosum_mod.F90" "-qsmp=noauto:noomp")
+  e3sm_add_flags("${CIMESRC_PATH}/share/util/shr_reprosum_mod.F90" "-qsmp=noauto:noomp")
 endif()
 
 # These routines benefit from -qnostrict without violating the bfb test

